homeassistant/components/suez_water/sensor.py

```diff
@@ -51,45 +47,30 @@ def __init__(self, client: SuezClient, counter_id: int) -> None:
             manufacturer="Suez",
         )
 
-    def _fetch_data(self) -> None:
-        """Fetch latest data from Suez."""
-        try:
-            self.client.update()
-            # _state holds the volume of consumed water during previous day
-            self._attr_native_value = self.client.state
-            self._attr_available = True
-            self._attr_attribution = self.client.attributes["attribution"]
-
-            self._attr_extra_state_attributes["this_month_consumption"] = {}
-            for item in self.client.attributes["thisMonthConsumption"]:
-                self._attr_extra_state_attributes["this_month_consumption"][item] = (
-                    self.client.attributes["thisMonthConsumption"][item]
-                )
-            self._attr_extra_state_attributes["previous_month_consumption"] = {}
-            for item in self.client.attributes["previousMonthConsumption"]:
-                self._attr_extra_state_attributes["previous_month_consumption"][
-                    item
-                ] = self.client.attributes["previousMonthConsumption"][item]
-            self._attr_extra_state_attributes["highest_monthly_consumption"] = (
-                self.client.attributes["highestMonthlyConsumption"]
-            )
-            self._attr_extra_state_attributes["last_year_overall"] = (
-                self.client.attributes["lastYearOverAll"]
-            )
-            self._attr_extra_state_attributes["this_year_overall"] = (
-                self.client.attributes["thisYearOverAll"]
-            )
-            self._attr_extra_state_attributes["history"] = {}
-            for item in self.client.attributes["history"]:
-                self._attr_extra_state_attributes["history"][item] = (
-                    self.client.attributes["history"][item]
-                )
-
-        except PySuezError:
-            self._attr_available = False
-            _LOGGER.warning("Unable to fetch data")
-
-    def update(self) -> None:
-        """Return the latest collected data from Suez."""
-        self._fetch_data()
-        _LOGGER.debug("Suez data state is: %s", self.native_value)
+    @property
+    def native_value(self) -> float | None:
+        """Return the current daily usage."""
+        if self.coordinator.aggregated_data is None:
+            return None
```

Comment:
Okay so if you fixed the `_async_update_data` to return the type, this function is only called when the update succeeds, so then you can get rid of these if statements